monai/data/nifti_reader.py

```diff
@@ -135,3 +135,77 @@ def __getitem__(self, index):
                 continue
             compatible_meta[meta_key] = meta_datum
         return img, target, compatible_meta
+
+
+@export("monai.data")
+class NiftiDatasetd(Dataset):
+    """
+    Loads image/segmentation pairs of Nifti files from the given filename lists. Dict level transformations can be
+    specified for the dictionary data which is constructed by image, label and other metadata.
+    """
+
+    def __init__(self, image_files, seg_files=None, labels=None, as_closest_canonical=False, transform=None,
+                 image_only=True, dtype=None):
+        """
+        Initializes the dataset with the image and segmentation filename lists. The transform `transform` is applied
+        to the images and `seg_transform` to the segmentations.
+
+        Args:
+            image_files (list of str): list of image filenames.
+            seg_files (list of str): if in segmentation task, list of segmentation filenames.
+            labels (list or array): if in classification task, list of classification labels.
+            as_closest_canonical (bool): if True, load the image as closest to canonical orientation.
+            transform (Callable, optional): dict transforms to excute operations on dictionary data.
+            image_only (bool): if True return only the image volume, other return image volume and header dict.
+            dtype (np.dtype, optional): if not None convert the loaded image to this data type.
+        """
+
+        if len(image_files) != len(seg_files):
+            raise ValueError('Must have same number of image and segmentation files')
+
+        self.image_files = image_files
+        self.seg_files = seg_files
+        self.labels = labels
+        self.as_closest_canonical = as_closest_canonical
+        self.transform = transform
+        self.image_only = image_only
+        self.dtype = dtype
+
+    def __len__(self):
+        return len(self.image_files)
+
+    def __getitem__(self, index):
+        meta_data = None
+        if self.image_only:
+            img = load_nifti(self.image_files[index], as_closest_canonical=self.as_closest_canonical,
+                             image_only=self.image_only, dtype=self.dtype)
+        else:
+            img, meta_data = load_nifti(self.image_files[index], as_closest_canonical=self.as_closest_canonical,
+                                        image_only=self.image_only, dtype=self.dtype)
+        target = None
+        if self.seg_files is not None:
+            target = load_nifti(self.seg_files[index])
+        elif self.labels is not None:
```

Comment:
here we use 'label' key to represent both classification label and segmentation map. how about we use separate keys (for example 'seg' for segmentation arrays, 'label' for classification target. so that the user can easily load label and segmentation into the same output dict (this can be very useful for multitask learning)

Reply:
modified in the latest patch.
Thanks.